sor/src/main/java/com/bazaarvoice/emodb/sor/core/PendingCompaction.java

```diff
@@ -23,12 +24,12 @@
     private final UUID _initialCutoffId;
     private final Delta _delta;
     private final Delta _startingDelta;
-    private final List<UUID> _keysToDelete;
-    private final List<UUID> _compactionKeysToDelete;
+    private final List<DeltaClusteringKey> _keysToDelete;
+    private final List<DeltaClusteringKey> _compactionKeysToDelete;
     private final List<Map.Entry<UUID, Delta>> _deltasToArchive;
 
-    public PendingCompaction(UUID compactionKey, Compaction compaction, UUID changeId, UUID initialCutoffId, Delta delta, Delta startingDelta, List<UUID> keysToDelete,
-                             List<UUID> compactionKeysToDelete, List<Map.Entry<UUID, Delta>> deltasToArchive) {
+    public PendingCompaction(UUID compactionKey, Compaction compaction, UUID changeId, UUID initialCutoffId, Delta delta, Delta startingDelta, List<DeltaClusteringKey> keysToDelete,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS00107)